lib/src/data/datasource/remote/api_service.dart

```diff
@@ -0,0 +1,76 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+
+import '../../models/genre_response.dart';
+import '../../models/movie_response.dart';
+import 'i_api_service.dart';
+
+class ApiService implements IApiService {
```

Comment:
If you notice in this file you are making the same call for getPopular, getNowPlaying, getTopRated, getUpcoming. You can made something like this, to avoid the repetition of code
```
@override
  Future<MovieResponse> getMoviesByType({required String endpoint}) async {
    final response = await http
        .get(Uri.parse(_baseUrl + endpoint + '?api_key=' + _apiKey));
    if (response.statusCode == 200) {
      return MovieResponse.fromJson(jsonDecode(response.body));
    } else {
      throw Exception(moviesException);
    }
  }
```

Reply:
done